executor/analyze.go

```diff
@@ -1291,15 +1331,19 @@ type samplingMergeResult struct {
 	err       error
 }
 
-func (e *AnalyzeColumnsExec) subMergeWorker(resultCh chan<- *samplingMergeResult, taskCh <-chan []byte, l int, isClosedChanThread bool) {
+func (e *AnalyzeColumnsExec) subMergeWorker(
+	resultCh chan<- *samplingMergeResult,
+	taskCh <-chan []byte,
+	l int, idx int) {
+	isClosedChanThread := idx == 0
 	defer func() {
 		if r := recover(); r != nil {
 			buf := make([]byte, 4096)
 			stackSize := runtime.Stack(buf, false)
 			buf = buf[:stackSize]
 			logutil.BgLogger().Error("analyze worker panicked", zap.String("stack", string(buf)))
 			metrics.PanicCounter.WithLabelValues(metrics.LabelAnalyze).Inc()
-			resultCh <- &samplingMergeResult{err: errAnalyzeWorkerPanic}
+			resultCh <- &samplingMergeResult{err: getAnalyzePanicErr(r)}
 		}
 		// Consume the remaining things.
 		for {
```

Comment:
Do we need to subtract the memory usage of the remaining things coming from `taskCh`? Otherwise we may think there may be some memory leak in analyze.

Reply:
Never mind. We set consumed memory to zero when `colExec.buildSamplingStats` returns error.